### PR TITLE
use eos_token_id if there is no pad_token_id to avoid sending None

### PR DESCRIPTION
**Pull request by wukaixingxp**
Some models, like Llama models, did not have `pad_token_id `thus this [pad_token()](https://github.com/meta-pytorch/torchforge/blob/main/apps/grpo/main.py#L269) will return None, which cause the training failure.  This PR will use `eos_token_id` if there is no `pad_token_id` to avoid that.

### CODE REVIEW

**Comment by felipemello1** [comment]
hey @wukaixingxp , thanks for the PR! The code changes seem minimal, but i am skeptical. The llama model was tested multiple times. Why would the error only appear now? Was it because previously the context was too small and it was never padded?

Could it be that there is another root cause and this PR is only fixing the symptom?

**Comment by wukaixingxp** [comment]
previously claude provide my a '[workaround](https://github.com/wukaixingxp/forge/blob/julia/apps/coding-grpo/main.py#L158)' that can get my llama 8B training working but it is WRONG!! My grpo loss is constantly ± 1k ~ 180K. With this fix, my grpo loss is about ± 1 ..

**Comment by felipemello1** [comment]
@wukaixingxp cool, thank you, nice finding! Before i merge, do you mind posting one of your losses, does it go down?

**Comment by wukaixingxp** [comment]
Screenshot 2025-11-08 at 2 41 56 PM
Screenshot 2025-11-08 at 2 40 43 PM
It could be my problem that the reward and loss in the graph is still not looking good, but now the grpo loss is at normal range.

**Review by felipemello1 — APPROVED** [review]
